all bot test fail on Ubuntu 22.04 because of differing python version string

for example:
```
[  213s] AssertionError: Regex didn't match: 'ShadowserverJSONParserBot initialized with id test-bot and intelmq [0-9a-z.]* and python [0-9a-z.]{5,8}\\+? \\([a-zA-Z0-9,:. ]+\\)( \\[GCC\\])? as process [0-9]+\\.' not found in 'ShadowserverJSONParserBot initialized with id test-bot and intelmq 3.0.2 and python 3.10.4 (main, Apr  2 2022, 09:04:19) [GCC 11.2.0] as process 10051.'
```
applies to all bot tests of course
`( \\[GCC\\])?` needs to be extend to also hold the gcc version (or a non-greedy catch-all)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -100,6 +100,7 @@ CHANGELOG
 - Add GitHub Action to run regexploit on all Python, JSON and YAML files (PR#2059 by Sebastian Wagner).
 - `intelmq.lib.test`:
   - Decorator `skip_ci` also detects `dpkg-buildpackage` environments by checking the environment variable `DEB_BUILD_ARCH` (PR#2123 by Sebastian Wagner).
+  - Fixing regex to catchall after python version and process id (PR#2216 by Sebastian Waldbauer, fixes #2185)
 - Also test on Python 3.10 (PR#2140 by Sebastian Wagner).
 - Switch from nosetests to pytest, as the former does not support Python 3.10 (PR#2140 by Sebastian Wagner).
 - CodeQL Github Actions `exponential backtracking on strings` fixed. (PR#2148 by Sebastian Waldbauer, fixes #2138)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -100,7 +100,7 @@ CHANGELOG
 - Add GitHub Action to run regexploit on all Python, JSON and YAML files (PR#2059 by Sebastian Wagner).
 - `intelmq.lib.test`:
   - Decorator `skip_ci` also detects `dpkg-buildpackage` environments by checking the environment variable `DEB_BUILD_ARCH` (PR#2123 by Sebastian Wagner).
-  - Fixing regex to catchall after python version and process id (PR#2216 by Sebastian Waldbauer, fixes #2185)
+  - Fixing regex to catchall after python version and process ID, add tests for it (PR#2216 by Sebastian Waldbauer and Sebastian Wagner, fixes #2185)
 - Also test on Python 3.10 (PR#2140 by Sebastian Wagner).
 - Switch from nosetests to pytest, as the former does not support Python 3.10 (PR#2140 by Sebastian Wagner).
 - CodeQL Github Actions `exponential backtracking on strings` fixed. (PR#2148 by Sebastian Waldbauer, fixes #2138)

--- a/intelmq/lib/test.py
+++ b/intelmq/lib/test.py
@@ -44,6 +44,11 @@ BOT_CONFIG = {"destination_pipeline_broker": "pythonlist",
               "source_pipeline_broker": "pythonlist",
               "testing": True,
               }
+BOT_INIT_REGEX = ("{} initialized with id {} and"
+                  r" intelmq [0-9a-z.]*"
+                  r" and python [0-9a-z.]*"
+                  r" \(.*?\)([0-9a-zA-Z.\ \[\]]*)"
+                  r"as process [0-9]+\.")
 
 
 class Parameters:
@@ -348,13 +353,8 @@ class BotTestCase:
                 self.assertIn('raw', event)
 
         """ Test if bot log messages are correctly formatted. """
-        self.assertLoglineMatches(0, "{} initialized with id {} and"
-                                     r" intelmq [0-9a-z.]*"
-                                     r" and python [0-9a-z.]*"
-                                     r" \(.*?\)([0-9a-zA-Z.\ \[\]]*)"
-                                     r"as process [0-9]+\."
-                                     "".format(self.bot_name,
-                                               self.bot_id), "INFO")
+        self.assertLoglineMatches(0, BOT_INIT_REGEX.format(self.bot_name,
+                                                           self.bot_id), "INFO")
         self.assertRegexpMatchesLog("INFO - Bot is starting.")
         if stop_bot:
             self.assertLoglineEqual(-1, "Bot stopped.", "INFO")

--- a/intelmq/lib/test.py
+++ b/intelmq/lib/test.py
@@ -348,9 +348,11 @@ class BotTestCase:
                 self.assertIn('raw', event)
 
         """ Test if bot log messages are correctly formatted. """
-        self.assertLoglineMatches(0, "{} initialized with id {} and intelmq [0-9a-z.]* and python"
-                                     r" [0-9a-z.]{{5,8}}\+? \(.+?\)( \[GCC.*?\])?"
-                                     r" as process [0-9]+\."
+        self.assertLoglineMatches(0, "{} initialized with id {} and"
+                                     r" intelmq [0-9a-z.]*"
+                                     r" and python [0-9a-z.]*"
+                                     r" \(.*?\)([0-9a-zA-Z.\ \[\]]*)"
+                                     r"as process [0-9]+\."
                                      "".format(self.bot_name,
                                                self.bot_id), "INFO")
         self.assertRegexpMatchesLog("INFO - Bot is starting.")

--- a/intelmq/tests/lib/test_test.py
+++ b/intelmq/tests/lib/test_test.py
@@ -1,0 +1,25 @@
+# SPDX-FileCopyrightText: 2022 Intevation GmbH
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+# -*- coding: utf-8 -*-
+
+import unittest
+
+from intelmq.lib.test import BOT_INIT_REGEX
+
+
+BOT_INIT_LINES = (
+    # Ubuntu 22.04 https://github.com/certtools/intelmq/issues/2185
+    'BOTNAME initialized with id BOTID and intelmq 3.0.2 and python 3.10.4 (main, Apr  2 2022, 09:04:19) [GCC 11.2.0] as process 10051.',
+)
+
+
+class TestBotTestCase(unittest.TestCase):
+    def test_bot_init_regex(self):
+        for line in BOT_INIT_LINES:
+            self.assertRegex(line, BOT_INIT_REGEX.format('BOTNAME', 'BOTID'))
+
+
+if __name__ == '__main__':  # pragma: no cover
+    unittest.main()

